patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -791,13 +791,32 @@
 +     * @param armor the armor itemstack
 +     */
 +    public void onHorseArmorTick(World world, net.minecraft.entity.EntityLiving horse, ItemStack armor) {}
++    
++    @SideOnly(Side.CLIENT)
++    @Nullable
++    private net.minecraft.client.renderer.tileentity.TileEntityItemStackRenderer teisr = null;
++    
++    /**
++     * @return This Item's renderer, or the default instance if it does not have one.
++     */
++    @SideOnly(Side.CLIENT)
++    public final net.minecraft.client.renderer.tileentity.TileEntityItemStackRenderer getTileEntityItemStackRenderer()
++    {
++    	return teisr != null ? teisr : net.minecraft.client.renderer.tileentity.TileEntityItemStackRenderer.field_147719_a;
++    }
++    
++    @SideOnly(Side.CLIENT)
++    public void setTileEntityItemStackRenderer(@Nullable net.minecraft.client.renderer.tileentity.TileEntityItemStackRenderer teisr)
```

Comment:
Can you make this `protected` so that you can only change your own item's renderer?

Reply:
Protected means that it would have to be called from either the package containing your item class or the class itself, i'd prefer it throw an exception if the renderer is not null.

Reply:
If you want different access, you can override it in your `Item` method. The idea is just to make it harder to really mess up something else's rendering.